vars/piperPipelineStageConfirm.groovy

```diff
@@ -0,0 +1,80 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateStageDocumentation
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = [
+    /**
+     * Specifies if a manual confirmation is active before running the __Promote__ and __Release__ stages of the pipeline.
+     * @possibleValues `true`, `false`
+     */
+    'manualConfirmation',
+    /** Defines message displayed as default manual confirmation. Please note: only used in case pipeline is in state __SUCCESSFUL__ */
+    'manualConfirmationMessage',
+    /** Defines how many hours a manual confirmation is possible for a dedicated pipeline. */
+    'manualConfirmationTimeout'
+
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * In this stage a manual confirmation is requested before processing subsequent stages like __Promote__ and __Release__.
+ *
+ * This stage will be active in two scenarios:
+ * - manual activation of this stage
+ * - in case of an 'UNSTABLE' build (even when manual confirmation is inactive)
+ */
+@GenerateStageDocumentation(defaultStageName = 'Confirm')
+void call(Map parameters = [:]) {
+    def script = checkScript(this, parameters) ?: this
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    Map config = ConfigurationHelper.newInstance(this)
+        .loadStepDefaults()
+        .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+        .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
+        .mixin(parameters, PARAMETER_KEYS)
+        .use()
+
+    String unstableStepNames = script.commonPipelineEnvironment.getValue('unstableSteps') ? "${script.commonPipelineEnvironment.getValue('unstableSteps').join(':\n------\n')}:" : ''
+
+    boolean approval = false
+    def userInput
+
+    timeout(
+        unit: 'HOURS',
+        time: config.manualConfirmationTimeout
+    ){
+        if (currentBuild.result == 'UNSTABLE') {
+            while(!approval) {
+                userInput = input(
```

Comment:
we should offer the use of the `submitter` attribute to limit the users who can approve